site/tests/app/controllers/HomePageControllerTester.php

```diff
@@ -0,0 +1,97 @@
+<?php
+
+namespace tests\app\controllers;
+
+use app\libraries\homework\UseCases\LibraryAddUseCase;
+use app\libraries\response\RedirectResponse;
+use app\libraries\response\WebResponse;
+use app\libraries\response\JsonResponse;
+use app\models\User;
+use app\libraries\Core;
+use tests\BaseUnitTest;
+use app\controllers\HomePageController;
+use app\libraries\FileUtils;
+
+class HomePageControllerTester extends BaseUnitTest {
+
+	protected $controller;
+
+	protected $core;
+
+	protected $tmp_dirs = ["DAEMON_CONTAINED_WITHIN", "LIVE_UPDATES", "THERE_SHALL_BE_LIGHT", "THIS_IS_ALSO_IMPORTANT", "WE_WANT_TO_SEE_THIS"];
+
+	protected $library_path;
+
+	public function tearDown(): void {
+		FileUtils::recursiveRmdir($this->library_path);
+	}
+
+	public function setUp(): void {
+		parent::setUp();
+
+		$this->library_path = FileUtils::joinPaths(sys_get_temp_dir(), "library");
+
+		$this->core = $this->createMockCore([
+			'getUsernameChangeText' => True,
+            'homework_library_enable' => True,
+            'homework_library_location' => '/tmp/library'
+		], [
+			'accessGrading' => True
+		], [
+			'updateUser' => []
+		], [], [
+			'addSuccessMessage' => true,
+			'addErrorMessage' => true,
+			'buildUrl' => 'http://192.168.56.111/home'
```

Comment:
Why do you need to specify this in the test?

Reply:
Ah this is archaic code... 😳